new/supermarket_namespace_governance.md

```diff
@@ -0,0 +1,167 @@
+    
+In this model, we draw parallels with programming languages. In most
+language communities, there is a system for contributing  and sharing
+libraries. Perl has CPAN. Ruby has Rubygems.org, Python has PyPI. Node 
+has NPM, and Java has Maven. Haskell has Hackage. Erlang has Hex.
+    
+There is a ton of momentum behind this mental model. Tooling such as
+Berkshelf was built with this in mind. Cookbooks are packages and the
+Supermarket is the hosting system.
+   
+Namespace governance is typically optimized for the preservation of
+backwards compatibility of individual components consumed by build
+systems.
+    
+When two libraries are concerned with the same topic, but work in
+different ways, they play semantic games to distinguish themselves.
+
+Some languages, by convention, use a special character for this
+purpose. 'alice/json' vs 'bob/json' is a popular method. Enabling this
+requires coordination among the ecosystem tooling at every level  and
+implementations vary wildly. It would be worth exploring the logistics
+of around this, if for no other reason than to have something to point
+to when people suggest it.
+
+Without a special character, semantic strategies fall on a spectrum.
+At one end there is 'json' vs 'json2', or 'json-alice' vs 'json-bob'.
+At the other end, you have... Ruby. While it may be perfectly
+acceptable in the Ruby community to name an XML processing library
+after a Japanese tree saw, I suspect that the Chef community might
+prefer something less random. A governance activity might include the
+creation and enforcement of a recommended naming conventions.
+   
+Some languages have the notion of "core modules", where different
+governance rules come into play. This feels like something that Chef
+should adopt.
+
+This model starts to show cracks when you consider that most Chef
+cookbooks currently on the supermarket are not written in a reusable
+manner. Cookbooks that provide resources can be considered libraries
+or modules, but recipes are really opinionated state policies.
+
+### Chef as a distribution
+
+In this model, we draw parallels with Linux and Unix distributions.
+Most distributions can be described as the set of available packages
+and the metadata that binds them.
+
+Namespace governance is typically optimized for content curation, with
+an emphasis on interoperability. To achieve this, packaging policies
+are created to enforce common conventions.
+
+This model is often used by sophisticated Chef shops, who often write
+everything themselves and rarely consume cookbooks from the public
+Supermarket.
+   
+In Linux distributions, two packages are rarely, if ever, concerned
+with the same software. Rather than having 'alice/apache' vs
+'bob/apache', the distribution will pick a preferred package and
+declare if "the way of the system". Often these packages vary wildly
+between distributions. The configuration conventions around the Apache
+httpd server is a great example of this.
```

Comment:
I feel like I've seen a lot of counter examples to this, e.g. supporting multiple PHP versions that have been namespaced using package names, as well as sun-java/openjdk and various latex distributions. I'm not disagreeing with your sentiment, but we're likely to have similar outcomes if we declare a 'one true way' model. We've been rather lucky that, for example, the java cookbook has supported both flavors (sun/openjdk). I'm not sure every cookbook would end up with that much unity.